### PR TITLE
Median construct correlations plate 4

### PR DESCRIPTION
**Pull request by MattsonCam**
This analysis finds the aggregate correlation pdfs across wells for both the same and different siRNA constructs with some refactoring. The filtering operations to compute the correlation distributions are complex, so feel free to ask any questions. Also, further analysis is warranted, and is to be continued.

### CODE REVIEW

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/correlation_relationships/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Correlation Analysis
+In this analysis, the features values were compared for all cells for the following groups:
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/correlation_relationships/README.md]:
```suggestion
In this analysis, the features values were compared for all cells.
```

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+# ## Imports
+
+# In[1]:
+
+
+import pandas as pd
+from pathlib import Path
+import sys
+
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py]:
```suggestion
import sys
from pathlib import Path

import pandas as pd

import create_correlations as cc
```
Order should be:  
1. Standard library imports.
2. Related third party imports.
3. Local application/library specific imports.  
As seen in [PEP8](https://peps.python.org/pep-0008/)

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# ## Import helper correlation class
+
+# In[3]:
+
+
+sys.path.append(
+    f"{root_dir}/0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils"
+)
+import create_correlations as cc
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py:48]: Consider importing at beginning of the Notebook

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+corr_obj = cc.CreateCorrelations(platedf=platedf, aggregate=True)
+
+
+# ## Create correlation pdfs
+
+# In[7]:
+
+
+corr_obj.plot_correlations(output_path=output_path)
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py]: blank line at end of script. Consider using black to format these

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
+import pandas as pd
+import numpy as np
+from collections import defaultdict
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py:1]: same here see above comment about PEP8 import order.

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+import matplotlib.pyplot as plt
+from scipy.stats import gaussian_kde
+
+
+class CreateCorrelations:
+    def __init__(self, platedf, aggregate):
+        """
+        Parameters
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py:12]: Really good documentation throughout this class for methods and class thank you for making this clean

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+            # Keep track of the wells used for the correlation
+            tried_wells.append(well)
+
+        # Combine the correlation data
+        corr_data["comparison"] = [comp] * len(corr_data["first_well"])
+        corr_data["pearsons_coef"] = np.concatenate(corr_data["pearsons_coef"])
+
+        return pd.DataFrame(corr_data)
+
+    def plot_correlations(self, output_path):
```
Comment (MikeLippincott) [inline 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py:208]: missing a docstring here. Consider adding one

**Review by MikeLippincott — APPROVED** [review]
A few minor documentation fixes and styling fixed and ready to merge. Nice PR!

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/correlation_relationships/README.md (hunk 1 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/correlation_relationships/README.md]: Changed

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py (hunk 2 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py]: Sorted

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py (hunk 4 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py]: I formatted the jukit python file with black, but not the nbconverted file.

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py (hunk 5 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py:1]: Sorted

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py (hunk 7 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/utils/create_correlations.py:208]: Good catch

File: 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py (hunk 3 above)
Comment (MattsonCam) [reply 0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/median_correlation_relationships/nbconverted/median_cross_corr.py:48]: I thought about that, but I would have to define the block of code where I create the `root_dir` path above the imports to maintain that structure, which also relies on imports.

**Comment by MattsonCam** [comment]
Thanks Mike, merging now!

**Pull request merged**